src/Microsoft.FeatureManagement/VariantConfigurationSection.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.FeatureManagement
+{
+    internal class VariantConfigurationSection : IConfigurationSection
+    {
+        private readonly ConfigurationRoot _root;
+        private readonly string _path;
+        private string _key;
+
+        public VariantConfigurationSection(string key, string path, string value)
+        {
+            MemoryConfigurationSource source = new MemoryConfigurationSource();
+
+            _root = new ConfigurationRoot(new List<IConfigurationProvider> { new MemoryConfigurationProvider(source) });
```

Comment:
I don't understand why _root is needed.

Reply:
We have to implement the `this[string key]` operator, so if a user tries to do that on this VariantConfigurationSection I wasn't sure if just making it a no-op was better than allowing them to use it like a regular ConfigurationSection.

Reply:
Even if we did want to, we don't need configuration root to do it. It looks like a dictionary could suffice to fulfill that contract. 

I feel like it might be better for us not to implement IConfigurationSection indexer setter.